multicluster/controllers/multicluster/member/node_controller.go

```diff
@@ -199,7 +199,10 @@ func (r *NodeReconciler) updateActiveGateway(ctx context.Context, newGateway *mc
 	// TODO: cache might be stale. Need to revisit here and other reconcilers to
 	// check if we can improve this with 'Owns' or other methods.
 	if err := r.Client.Get(ctx, types.NamespacedName{Name: newGateway.Name, Namespace: r.namespace}, existingGW); err != nil {
-		return err
+		if apierrors.IsNotFound(err) {
+			r.activeGateway = ""
+		}
+		return client.IgnoreNotFound(err)
```

Comment:
We can just return nil in line 204?

Reply:
Both works, client.IgnoreNotFound(err) will also return nil if it's not found error. Changed to following codes:
```
		if apierrors.IsNotFound(err) {
			r.activeGateway = ""
			return nil
		}
		return err
```